File: src/c/ryu/tests/dfmt_echo.c
// Copyright 2018 Ulf Adams
//
// The contents of this file may be used under the terms of the Apache License,
// Version 2.0.
//
//    (See accompanying file LICENSE-Apache or copy at
//     http://www.apache.org/licenses/LICENSE-2.0)
//
// Alternatively, the contents of this file may be used under the terms of
// the Boost Software License, Version 1.0.
//    (See accompanying file LICENSE-Boost or copy at
//     https://www.boost.org/LICENSE_1_0.txt)
//
// Unless required by applicable law or agreed to in writing, this software
// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.

#if defined(__linux__)
#define _GNU_SOURCE
#endif

#include <stdbool.h>
#include <inttypes.h>
#include <time.h>
#include <string.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

#if defined(__linux__)
#include <sys/types.h>
#include <unistd.h>
#endif

#include "ryu/ryu.h"
#include "ryu/ryu_parse.h"

int main (int argc, char **argv) {
    char buffer[1024];
    double d;
    while (fgets(buffer, 1024, stdin) != NULL) {
        buffer[strcspn(buffer,"\n")] = 0;
        if (dparse(buffer, &d) == SUCCESS) {
            dfmt(d, buffer);
            puts(buffer);
        } else {
            puts("ERROR");
        }
    }
}

File: src/java/testsrc/com/ironmeld/doubleback/DfmtEcho.java
// Copyright 2018 Ulf Adams
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.ironmeld.doubleback;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

class DfmtEcho {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String nextValue;
        while ((nextValue = br.readLine()) != null) {
            try {
                double f = Doubleback.dparse(nextValue);
                System.out.println(Doubleback.dfmt(f));
            }
            catch (IllegalArgumentException e) {
                System.out.println("ERROR");
            }
        }
    }
}
